src/util/LockUtils.ts

```diff
@@ -0,0 +1,102 @@
+import { getLoggerFor } from '../logging/LogUtil';
+import { InternalServerError } from './errors/InternalServerError';
+
+const logger = getLoggerFor('LockUtil');
+
+/**
+ * Await this function to wait a set amount of time (without consuming cpu).
+ * @param delay - How long to wait.
+ * @param jitter - A fraction of this jitter will be added to the delay.
+ * @returns Promise<void> when ready.
+ */
+export async function setJitterTimeout(delay: number, jitter = 0): Promise<void> {
+  function calcTime(): number {
+    jitter = jitter > 0 ? Math.max(1, Math.floor(Math.random() * jitter)) : 0;
+    return Math.max(0, delay + jitter);
+  }
+  return new Promise<void>((resolve): any => setTimeout(resolve, calcTime()));
```

Comment:
The nesting was because of previous refactoring where it was useful. 
Actually the 1 is because it was easier to test adding jitter in my tests. I can tell jest to advance time to `delay + max jitter` which should have resolved. The measured time until resolved should than be greater than `delay` (becuase of at least 1ms being added).
https://github.com/CommunitySolidServer/CommunitySolidServer/blob/a1fa06220f727e4afa6058eef6311706ffe2079a/test/unit/util/LockUtil.test.ts#L18-L26

Reply:
The trick is to mock `Math.random` with jest so you know exactly how much the jitter will be.